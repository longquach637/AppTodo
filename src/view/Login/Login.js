import React from "react";
import { useState } from "react";

export default function Login(props) {
  const { email: propsEmail, password: propsPassword, dispatch } = props;
  const [isLoginDisabled, setIsLoginDisabled] = React.useState(true);
  const [email, setEmail] = React.useState(propsEmail || "");
  const [password, setPassword] = React.useState(propsPassword || "");

  React.useEffect(() => {
    validateForm();
  }, [email, password]);

  const validateEmail = (text) => /@/.test(text);

  const validateForm = () => {
    setIsLoginDisabled(password.length < 8 || !validateEmail(email));
  };

  const handlePasswordChange = (evt) => {
    const passwordValue = evt.target.value.trim();
    setPassword(passwordValue);
  };

  const handleSubmit = () => {
    dispatch("submit(email, password)");
    setIsLoginDisabled(true);
  };

  const handleDelete = () => {};

  const [counter, setcounter] = useState(0);

  return (
    <form>
      <span className="mx-auto" />
      <input
        type="password"
        id="my-2"
        onChange={handlePasswordChange}
        value={password}
      />
      <input
        type="button"
        className="btn btn-primary"
        onClick={handleSubmit}
        disabled={isLoginDisabled}
        value="Submit"
      />
      <button onClick={handleDelete}>Delelte</button>
      <div>
        <h1>This is counter app</h1>
        <div id="counter-value">{counter}</div>
        <button id="increment-btn" onClick={() => setcounter(counter + 1)}>
          Increment
        </button>
        <button id="decrement-btn" onClick={() => setcounter(counter - 1)}>
          Decrement
        </button>
      </div>
    </form>
  );
}
