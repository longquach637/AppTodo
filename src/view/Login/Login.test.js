import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, shallow } from "enzyme";
import { reduce } from "lodash";
import React from "react";
import Login from "./Login";

configure({ adapter: new Adapter() });

describe("<Login /> with no props", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Login />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should find input", () => {
    expect(wrapper.find("#my-2").length).toEqual(1);
  });

  afterEach(() => {
    wrapper.unmount();
  });
});

describe("Counter Testing", () => {
  let wrapper;
  let state;
  beforeEach(() => {
    wrapper = shallow(<Login />);
    state = {
      counter: 0,
    };
  });

  test("render the title of counter", () => {
    expect(wrapper.find("h1").text()).toContain("This is counter app");
  });

  test("render a button with text of `increment`", () => {
    expect(wrapper.find("#increment-btn").text()).toBe("Increment");
  });

  test("render the initial value of state in a div", () => {
    expect(Number(wrapper.find("#counter-value").text())).toBe(state.counter);
  });

  test("render the click event of increace button and decrement counter value", () => {
    wrapper.find("#increment-btn").simulate("click");
    expect(wrapper.find("#counter-value").text()).toBe("1");
  });

  test("render the click event of decrement button and decrement counter value", () => {
    wrapper.find("#increment-btn").simulate("click");
    expect(wrapper.find("#counter-value").text()).toBe("1");

    wrapper.find("#decrement-btn").simulate("click");
    expect(wrapper.find("#counter-value").text()).toBe("0");
  });

  afterEach(() => {
    wrapper.unmount();
  });
});
