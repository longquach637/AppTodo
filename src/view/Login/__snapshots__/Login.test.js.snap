// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> with no props should match the snapshot 1`] = `ShallowWrapper {}`;
